test/faye-redis.unit.js

```diff
@@ -0,0 +1,71 @@
+const { expect } = require('chai');
+const faye = require('faye');
+const hat = require('hat');
+const redis = require('redis');
+
+const fayeRedis = require('../faye-redis');
+const utils = require('./utils');
+
+describe('Redis engine', function() {
+    const namespace = 'faye';
+    let channel;
+    let clientId;
+    let engineOpts;
+    let redisClient;
+    let engine;
+    let bayeux;
+
+    before(() => {
+        redisClient = redis.createClient();
+    });
+
+    after(async () => {
+        await utils.invokeAsPromise(redisClient, 'quit');
+    });
+
+    beforeEach(() => {
+        bayeux = new faye.NodeAdapter({ mount: '/faye', timeout: 45 });
+        engine = fayeRedis.create(bayeux._server._engine, { namespace });
+
+        clientId = hat();
+        channel = hat();
+    });
+
+    afterEach(() => {
+        engine.disconnect();
+        bayeux.close();
+    });
+
+    describe('#subscribe', function() {
+        it('should subscribe the client to the channel', async () => {
+            engine.subscribe(clientId, channel);
+            await utils.delay(500);
```

Comment:
What's the delay for?

Reply:
The subscribe is asynchronous and if there is no delay the tests might run too fast and not succeed.

Reply:
>subscribe is asynchronous

Redis methods accept a callback, so you could:

```
await new Promise((resolve) => engine.subscribe(clientId, channel, resolve));
```

Reply:
Didn't see the callback option... thanks! I'll update the code.